client/my-sites/plans/current-plan/thank-you-header/index.js

```diff
@@ -0,0 +1,90 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { get, isFinite, noop } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import getCurrentPlanPurchaseId from 'state/selectors/get-current-plan-purchase-id';
+import ProgressBar from 'components/progress-bar';
+import QuerySitePurchases from 'components/data/query-site-purchases';
+import { getByPurchaseId } from 'state/purchases/selectors';
+import { getPlan } from 'lib/plans';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSitePlanSlug } from 'state/sites/selectors';
+
+export class PlanSetupHeader extends Component {
+	static propTypes = {
+		progressComplete: PropTypes.number,
+		progressTotal: PropTypes.number,
+	};
+
+	render() {
+		const { moment, purchaseExpiryDate, planName, siteId, translate } = this.props;
+		const duration = purchaseExpiryDate
+			? moment.duration( moment().diff( purchaseExpiryDate ) ).humanize()
+			: null;
+		return (
+			<Card className="thank-you-header">
+				<QuerySitePurchases siteId={ siteId } />
+				<img
+					className="thank-you-header__illustration"
+					alt=""
+					aria-hidden="true"
+					src="/calypso/images/illustrations/fireworks.svg"
+				/>
+				<h1 className="thank-you-header__title">{ translate( 'Thank you for your purchase!' ) }</h1>
+				<p>
+					{ duration && planName
+						? translate(
+								'Your website is on a %(planName)s plan for %(duration)s. Let’s walk through a short checklist of essential security features for safeguarding your website.',
+								{
+									args: { duration, planName },
+								}
+						  )
+						: ' ' /* &nbsp; maintain some space */ }
+				</p>
+				<p>
+					{ translate(
+						'We’ve taken the liberty of starting the first two items, since they’re key to your site’s safety: we’re configuring spam filtering and backups for you now. Once that’s done, we can work through the rest of the checklist.'
+					) }
+				</p>
+				{ /* can be 0 */ isFinite( this.props.progressComplete ) &&
+					/* shouldn't be 0 */ this.props.progressTotal && (
+						<ProgressBar
+							isPulsing
+							total={ this.props.progressTotal }
+							value={ this.props.progressComplete }
+						/>
+					) }
+				<p>
+					<a
+						className="thank-you-header__skip-setup"
+						href={ /* @TODO (sirreal) fix this */ document.location.pathname }
+					>
+						{ translate( 'Skip setup. I’ll do this later.' ) }
+					</a>
+				</p>
+			</Card>
+		);
+	}
+}
+
+export default connect( state => {
+	const siteId = getSelectedSiteId( state );
+	const planSlug = getSitePlanSlug( state, siteId );
+	return {
+		planName: get( getPlan( planSlug ), [ 'getTitle' ], noop )(),
+		purchaseExpiryDate: get( getByPurchaseId( state, getCurrentPlanPurchaseId( state, siteId ) ), [
+			'expiryDate',
+		] ),
```

Comment:
Several questions here:
* do we need extra security (`getByPurchaseId` doesn't seem to account for a purchase ID that's not valid)?
* do we need some caching?
* do you think this will make a good selector - `getCurrentPlanPurchase()`?
* nit: do we need square brackets?

Reply:
* > do we need extra security (`getByPurchaseId` doesn't seem to account for a purchase ID that's not valid)?

  The selector's current implementation:
   https://github.com/Automattic/wp-calypso/blob/87149d08dd4299add76d5c9860adfad5df3738a4/client/state/purchases/selectors.js#L52-L55
   ```
   > [ { id: "id" } ].filter( ( { id } ) => id === null ).shift()
   undefined
   > require( 'lodash' ).get( undefined, [ 'any', 'path' ] )
   undefined
   ```
   I don't see a problem at connect-level. Unless I've missed something, the rest of the component uses an `undefined` `purchaseExpiryDate` safely.

* > do we need some caching?

  I don't think so. Why do you ask?
* > do you think this will make a good selector - `getCurrentPlanPurchase()`?

  #27039 and d2db91b31a93fad772f4b0e44deff613852f5f6d
* > nit: do we need square brackets?
  
  https://github.com/Automattic/wp-calypso/pull/26881#discussion_r215539374

Reply:
> I don't think so. Why do you ask?

`getByPurchaseId()` will internally create a new array every time, and will throw it away, returning the first element. `getCurrentPlanPurchaseId()` will use the `getCurrentPlan()` which can create a new plan object every time. And we rely on both of these on every update, which felt a bit expensive to me (although I haven't tested the perf implications myself). What do you think?

I'm satisfied about the rest, thank you!

Reply:
A preface - these are questions for selectors which aren't introduced here. Ideally, we can rely on selectors being safe and performant enough to use at will.

I knew about the assembler used for purchases, which I hope to remove someday (#25391). The assembler in the purchases selector is cached:

https://github.com/Automattic/wp-calypso/blob/190db6dfe419228102bc110e2c1d86f91d369b93/client/state/purchases/selectors.js#L23-L26

I don't think the intermediate `shift . filter` is something to worry about (although ideally this would be `find`).

I wasn't aware of the assembler in plans, but the costly uncached assembler in only used as a fallback. The site plan data passes through the assembler when entering the app (on API response) and some quick testing indicates that this pre-assembled site plan is used here.

https://github.com/Automattic/wp-calypso/blob/80f01f0f9cb8216eed5cce533adf39139e6e8664/client/state/sites/plans/selectors.js#L39-L55

Again, I don't think this is something we need to worry about.

I tried to look at performance here as state loads, but it's tricky here because when there's no app state loaded, we're redirected from `/plans/my-plan` to `/plans` 🤷‍♂️

Reply:
Yeah, let's not dig further into this - thanks for investigating!